### PR TITLE
Replace unmaintained HammerJS dependency with VueUse

### PR DESCRIPTION
**Pull request by susnux**
HammerJS is unmaintained, ref: https://github.com/hammerjs/hammer.js/issues/1278

Replaced with VueUse which will be quite useful as we can also use it for #3893 so we can replace two unmaintained dependencies with one maintained one.

(Not relevant but this also will fix issues when building nextcloud-vue with vite (maybe in the future)).

### CODE REVIEW

**Review by Pytal — APPROVED** [review]

File: src/components/NcAppContent/NcAppContent.vue [hunk 1]
```diff
@@ -255,34 +259,36 @@ export default {
 	},
 
 	mounted() {
-		if (this.allowSwipeNavigation) {
```
[the remark this answers is not in the document]
Comment (susnux) [reply src/components/NcAppContent/NcAppContent.vue:258]: :+1: fixed

**Review by marcoambrosini — COMMENTED** [review]
I'm trying to test this from the web in mobile mode but it doesn't work for the app navigation

**Review by ShGKme — CHANGES_REQUESTED** [review]
I tried in Talk, and swipe for the left sidebar doesn't work for me now....
Nextcloud 27, Android, Vivaldi (Chromium-based).

Does it work for you in any app/mobile browser?

**Review by ShGKme — APPROVED** [review]
Tested on Android - Vivaldi (Chromium) and Firefox - works fine.

@marcoambrosini Could you also test on iPhone Safari?

**Review by marcoambrosini — APPROVED** [review]
Tested and works on safari mobile

**Review by raimund-schluessler — APPROVED** [review]

**Pull request merged**